code/modules/mob/living/simple_animal/hostile/cat_butcher.dm

```diff
@@ -33,90 +34,168 @@
+	if(L.blood_volume <= 500) //bandage them up and give em some blood if they're bleeding
+		L.blood_volume += 30
+		L.suppress_bloodloss(1800)
+	if(L.getBruteLoss() >= 50)
+		var/healing = min(L.getBruteLoss(), 120)
+		L.adjustBruteLoss(-healing)
+		L.suppress_bloodloss(1800)//bandage their ass
+	FindTarget()
+
+/mob/living/simple_animal/hostile/cat_butcherer/proc/newvictim(var/mob/living/carbon/human/L)
+	if(victims.Find(L))
+		adjustHealth(-(maxHealth))
+		return FALSE
+	if(L.mind)
+		victims += L
+		say(pick("I'm a genius!!", "KITTY!!", "Another successful experiment!!", "Substandard product.", "You had better not run off, now!", "You never cease to amaze me, me."))
+		if(LAZYLEN(victims) <= 10)
+			maxHealth = (100 + (20 * LAZYLEN(victims)))
+		else
+			maxHealth = (300 + (5 * (LAZYLEN(victims)-10)))
+		if(LAZYLEN(victims) == 2) 
+			projectiletype = /obj/item/projectile/bullet/dart/tranq/plus
+		if(LAZYLEN(victims) == 4)//gain space adaptation to make cheesing harder
+			atmos_requirements = list("min_oxy" = 0, "max_oxy" = 0, "min_tox" = 0, "max_tox" = 0, "min_co2" = 0, "max_co2" = 0, "min_n2" = 0, "max_n2" = 0)
+			icon_state = "cat_butcher_fire"
+			icon_living = "cat_butcher_fire"
+		if(LAZYLEN(victims) == 6) //at this point, it's probably out in the hall attacking several people at once
+			ranged_cooldown_time = 20
+			rapid_melee = 2
+			transform *= 1.25
+		if(LAZYLEN(victims) == 8)
+			projectiletype = /obj/item/projectile/bullet/dart/tranq/plusplus
+		if(LAZYLEN(victims) == 10)
+			ranged_cooldown_time = 10
+		if(LAZYLEN(victims) == 15)//if he's gotten this powerful, someone has really fucked up
+			ranged_cooldown_time = 5
+			rapid_melee = 3
+			transform *= 1.25
+			icon_state = "cat_butcher_super"
+			icon_living = "cat_butcher_super"
+	adjustHealth(-(maxHealth))
+
+//cat butcher ai shit
+/mob/living/simple_animal/hostile/cat_butcherer/CanAttack(atom/the_target)
+	if(iscarbon(target))
+		var/mob/living/carbon/human/C = target
+		if(C.getorgan(/obj/item/organ/ears/cat) && C.getorgan(/obj/item/organ/tail/cat) && C.has_trauma_type(/datum/brain_trauma/severe/pacifism))//he wont attack his creations
+			if(C.stat && (!HAS_TRAIT(C, TRAIT_NOMETABOLISM) || !istype(C.dna.species, /datum/species/ipc)))//unless they need healing
+				return ..()
+			else
+				return FALSE
+	return ..()
+
+/mob/living/simple_animal/hostile/cat_butcherer/MoveToTarget(list/possible_targets)
+	if(target)
+		if(ishuman(target))
+			var/mob/living/carbon/human/L = target
+			if(L.health <=30 || L.stat || !L.can_inject(null, FALSE)) // base health to move in to attack is 30, not 40, as it accounts for armor somewhat
+				retreat_distance = 0
+			else
+				retreat_distance = 3 //spam chems if they aren't low and can be injected
+		else 
+			retreat_distance = 0//dont shoot chems at mobs that dont give a fuck
+	return ..()
+
+/mob/living/simple_animal/hostile/cat_butcherer/PickTarget(list/Targets)
+	if(target != null)
```

Comment:
```suggestion
	if(target)
```
Archanial suggested this before and they're right. It doesn't really matter if you copied it or not xD